### PR TITLE
CMR-5930: Support UMM-G 1.6

### PR DESCRIPTION
**Pull request by mschmele**

### CODE REVIEW

File: umm-spec-lib/src/cmr/umm_spec/migration/version/granule.clj [hunk 1]
```diff
@@ -61,6 +61,44 @@
...
+                          :SizeInBytes)
+
+      (util/update-in-all [:ArchiveAndDistributionInformation :FileType]
+                          dissoc
+                          :SizeInBytes)
+      (util/update-in-all [:ArchiveAndDistributionInformation :Files]
+                          dissoc
+                          :SizeInBytes)))
+
+(defn- safe-subs
```
[the remark this answers is not in the document]
Comment (mschmele) [reply umm-spec-lib/src/cmr/umm_spec/migration/version/granule.clj]: Found that there's actually a `trunc` function in the common-lib util file so I just used that instead

File: umm-spec-lib/src/cmr/umm_spec/migration/version/granule.clj [hunk 2]
```diff
@@ -61,6 +61,44 @@
...
+      (subs string begin end)
+      (subs string begin (count string)))))
+
+(defn- truncate-filename-type
+  "Migrate v1.6 FileNameType to v1.5 by truncating FilePackageType/Name and
+  FileType/Name fields to 80 characters or less"
+  [g]
+  (map (fn [id]
+         (-> id
+             (assoc :Identifier (safe-subs (:Identifier id) 0 79))
```
[the remark this answers is not in the document]
Comment (mschmele) [reply umm-spec-lib/src/cmr/umm_spec/migration/version/granule.clj]: Done

File: umm-spec-lib/test/cmr/umm_spec/test/migration/version/granule.clj [hunk 3]
```diff
@@ -202,3 +203,189 @@
...
+                                           :Size 11
+                                           :SizeUnit "KB"
+                                           :SizeInBytes 849324895784093
+                                           :Format "NETCDF-CF"
+                                           :FormatType "Supported"
+                                           :MimeType "application/x-netcdf"}]
+                                        :Identifiers
+                                         [{:Identifier "acos_LtCO2_090421_v201201_B7310A_161220013536s.nc4 and here are some extra characters so that this is too long for 1.5"
+                                           :IdentifierType "ProducerGranuleId"
+                                           :IdentifierName "Now this is the story all about how my life got flip-turned upside-down and I'd like to take a minute just sit right there"}]}
```
Comment (ygliuvt) [inline umm-spec-lib/test/cmr/umm_spec/test/migration/version/granule.clj]: Can you add a test case where the `IdentifierName` does not exist in v1.6? I don't think your code handles this case correctly.

Comment (mschmele) [reply umm-spec-lib/test/cmr/umm_spec/test/migration/version/granule.clj]: Fixed

**Pull request merged**